### PR TITLE
Rule 302: Items

### PR DESCRIPTION
**Pull request by Snark1994**
I introduce a new mechanic: items. Players can own items, which have some effect on the game. I've not allowed modification, introduction or removal of items: people are welcome to add these on later turns if they like. The only item I introduce is the _Top Hat of Good Karma_, which Peter is welcome to have.

### CODE REVIEW

File: rules.md [hunk 1]
```diff
@@ -106,4 +106,9 @@ This rule takes precedence over every other rule determining the winner except r
...
 
- * 301: Players take turns in alphabetical order by surname, starting with Nils B, taking one whole turn apiece. If a player joins or leaves the game, the current player must update `players.md`to the correct new turn order. A player may decide to either skip their turn, or carry it out without omitting any part of it. By omitting the first part of a turn, it is considered skipped and the next player begins their turn at the time they would if the turn had not been skipped.  All players begin with zero points.
+ * 301: Players take turns in alphabetical order by surname, starting with Nils B, taking one whole turn apiece. If a player joins or leaves the game, the current player must update `players.md` to the correct new turn order. A player may decide to either skip their turn, or carry it out without omitting any part of it. By omitting the first part of a turn, it is considered skipped and the next player begins their turn at the time they would if the turn had not been skipped.  All players begin with zero points.
+
+ * 302: Players may possess items, which are objects that can affect that state of the game.
```
Comment (cleborys) [inline rules.md]: "affect **that** state"

**Comment by cleborys** [comment]
I am a bit traumatized by people working together (malte's rule changing the vote system having philipp collect the mail addresses), so I would like the hat to be distributed randomly among everyone but Peter and Josh.
Although I don`t see how the hat could be abused, you never know...
I can propose a method to draw randomly shortly.

**Comment by Snark1994** [comment]
It's not collusion, I haven't mentioned it to Peter previously; I chose Peter because he complained that last round we never got round to making rules that weren't about rules. (I would also point out that the collusion was the thing I objected to in the last round's victory.) However, I don't really mind who gets it so if you want to nominate someone else then go ahead.

**Comment by cleborys** [comment]
I do believe you but would like to be careful.
To generate a random number I would add up the points scored on both sides for all four handball games scheduled [here](https://www.flashscore.com/handball/) on monday and consider them mod 4 with 0,1,2,3 identifying Nils, me, Malte, Philipp.

**Comment by Snark1994** [comment]
Can you not roll a die right now and tell me which of you 4 you would like to be the initial owner? (Or specify some other random event that will occur today, so it can be written up in time for tomorrow.) Otherwise I either need to put a messy bit of one-time handball-related rules explanation into the rules, or people will be voting for something that isn't exactly what will be committed.

**Comment by cleborys** [comment]
Sure. I rolled Nils.

**Comment by cleborys** [comment]
Could you change the last sentence of the rule to something like
`Unless specified in these rules or in the description of the item, items may neither change owner nor be destroyed or altered. Items me only be created by introduction in a new rule.`
and create the hat within your rule?

While it is to some extent already excluded in the rule, I would like to avoid shenanigans like the Wand of Winning, whose description states that it can be created at will and if so, I win.
It is "somewhat excluded" because `items.md` keeps track of descriptions, but I feel this could be interpreted either way and would like to avoid self-creating items.

**Comment by AnonymousVeNomicPlayer** [comment]
Is there a reason to prohibit the gerneration of items by other *already existing* items?

**Comment by Snark1994** [comment]
Philipp: would you accept me stating (here) that I intend the contents of `items.md` to reflect the state of items at the moment that the rule passes? (I think this is implicit in the rule itself, given it specifies that `items.md` must track the state of items - I see what you're getting at, but it seems a bit silly to add "The items are initially distributed according to `items.md`" given that is never going to be relevant again.)

**Comment by Snark1994** [comment]
And regarding other items: I considered phrasing it more generally, but I think items-generating-other-items is going to be pretty exploitable no matter how it's done. A similar effect could be obtained by having a "master" item M and some "slave" items S1, S2, ... SN that are in the bank, and M specifies how S1, ..., SN change ownership. Any item that creates another item would have to also specify how to create its description, which is essentially a way of introducing another rule. So I think it's safest to have all item generation done in advance via a rule-specified mechanism, and then items can transfer ownership if need be.

**Comment by AnonymousVeNomicPlayer** [comment]
I'm not sure. I agree that it is kind of silly, but I would just like to avoid someone claiming that the wall of winning already existed in the beginning or changing items.md a minute before midnight (it is not part of the rules before midnight). And I doubt that people must obey additions that are just mentioned in the comments.

**Comment by AnonymousVeNomicPlayer** [comment]
And can understand your concerns about items generating items.

**Comment by Snark1994** [comment]
If I put something in `rules.md` stating that the contents of `items.md` reflects the initial list of items, would that be enough? I can't see how your second point (changing `items.md` a minute before midnight on Sunday) could be addressed without an interpretation of the rules that states that `items.md` also has to be in its final form before Sunday, or specifying the full contents of `items.md` in the rule. I would rather not do the latter, and already follow the rules interpretation of the former.

**Comment by cleborys** [comment]
Bow before the scepter of item spam!

**Comment by cleborys** [comment]
I would have said that there are no starting items, unless you add the hat within your rule, since the only way to create an item now is within rules.

That is, the state of `items.md` is similarly to the rules independent of github?

**Comment by Snark1994** [comment]
Fine, I've explicitly added the Top Hat to the rules. Yes, the ownership and existence of items is independent of `items.md`.

**Comment by AnonymousVeNomicPlayer** [comment]
When you changed the rules, I was just about to write that I prefer a bit of spam in the rules over not having the hat introduced properly. I still wouldn't mind to have the description of the hat also explictly in the rules, if others have a similar opinion.

**Comment by malte-l** [comment]
Disclaimer: I could not look at this before, this is a quick comment so that things can still be changed before midnight.

I think if this rule got accepted, then there would be one item in existence, "Top Hat of Good Karma", which is in the bank (as it belongs to no one else) and has no associated properties or such things. According to 302, the ownership should be tracked in items.md, which however wrongly states that the owner is someone other than the bank. Furthermore I would find it confusing that the description in items.md pretends that there is a mechanism with the "Top Hat of Good Karma" that changes points, as there is no rule allowing something like this / giving things written in item.md this kind of power. I actually also believe the real description of the item would be empty if this rule-change is enacted as is, and item.md just has the wrong description. But even if the description would be the actual description, I think it still wouldn't be able to change points etc.

If there actually is the intention that the stuff written in the description should actually be what happens, then I think this has to either be a rule in itself, or there needs to be a rule stating clearly what kind of things descriptions in items.md can regulate and what takes precedence - rules or descriptions of items - etc.

Note also, that by 202 I think that we are only voting on the rule change, not the stuff  J. wants to put into item.md, so I think that even if the rule states that descriptions in item.md are as valid as rules, then this still doesn't make this description to be an enacted one. I believe there is no way around that what is part of the description has to be stated in a rule enacting the new description of the item at least once.

I propose that if J. wants to go ahead with items, then the first step should be to enact a rule stating things like that items exist, the thing with ownership, that they can't be created etc. except by rule and so on, and additionally how precisely they are created and how their effects are enacted and what priority they have, but without any initial item. That would have to be created in one of the following turns.

Let me add in the end that I am very skeptical of creating a way to introduce things with the supposed power of rule-changes (the descriptions) but kind of totally unregulated. This just screams like a loophole to be exploited somehow. I would advise anyone to vote "no" unless it is clear that description have no effect or there are strong rules in place regulating how those things can be enacted and changed and what power they have.

**Comment by malte-l** [comment]
I see that some aspects of this were already discussed a bit before, so let me add that I don't think that stuff in items.md is implicitly enacted by voting on this rule change. You can only change / create item.md *after* the rule has passed because otherwise this violates rule 122 and the rule-change refers to item.md (which is to be taken to mean the one in the master branch, where that file doesn't exist), not some proposal to change item.md after the rule passed or something like that.

**Comment by Snark1994** [comment]
Okay, I will completely and precisely specify everything about the item in `rules.md`, then. (I still don't feel this should be necessary.)

The rule already specifies that items can affect the state of the game; I will make that clearer.

There are the strongest possible rules in place regulating how items can be enacted and changed: it's explicitly forbidden to do either.

**Comment by malte-l** [comment]
The important part is not just enacting and changing, also what the descriptions can do. For example what happens if rules and descriptions contradict each other? I think the rules should always supersede the item descriptions and they should only be able to change stuff that is regulated by rules (such as points) if the rules explicitly state that. So for example if you want an item that gives points, then I would want a rule to need to exist that says that items can confer or deduct points, and perhaps it would then be a good idea to put a limit on how many on it.

**Comment by malte-l** [comment]
I don't see anything in the rule that rules out that only the person owning the item can "apply" the description. It rather seems to me that they can have arbitrary influence on the game, and if they explicitly state that they take precedence over all rules, then they do, perhaps even effectively circumventing rule 109 if that is actually a valid change. Though if this rule actually passes then I will think more about whether rule 110 actually makes this part of the proposed rule invalid, but it is not completely obvious to me if there is perhaps a loophole there that is being exploited here as well due to the assurances from the immutable  rules being too weak...
I don't yet see a method to exploit this to win as one person (of course, this doesn't mean that combined with other future changes there might not be a method where this is an essential part), but the potential risk of reducing "unanimity" to "majority" for effectively circumventing immutable rules seems dangerous to me.

**Comment by Snark1994** [comment]
The point about immutable rules is a good one; I will make it clear in the rule that items cannot overrule immutable rules.

**Comment by Snark1994** [comment]
I also don't think we necessarily need to always specify what happens if e.g. 2 emails are sent. Unless it's going to be game-breaking, I feel this is the kind of thing a judge would be able to sort out (and if I were asked to judge on it, I would be inclined to disincentivise people from doing annoying things like sending 2 emails).

The intention is for items to be at most another type of rule, and probably more limited in scope. I would expect most (all?) items created to only be usable by the owner. The aim was to allow more diversity and creativity in the range of actions that are possible within the game.

**Comment by AnonymousVeNomicPlayer** [comment]
I still want to say that I really appreciate the intention behind introducing the Top Hat of Good Karma, i.e. the idea to make the game more interesting again and to reward active players with some extra points.
I was afraid that after passing the rule that people do not need to propose rules any more and the ruling that no voting is as good as voting no, people do not need to participate in the game any more, meaning that by not proposing a rule they can't loose ten points if it does not get accepted and by their voting they can gain 10 points if a rule passes nevertheless. (Actually the length of my last sentence reminds me very much of reading Kleist in school ;))
Especially since it is now forbidden that an item overrules an immutable rule, I cannot see why they should be (too) dangerous (one should simply vote against a proposal to introduce a dangerous item). So I will approve the rule change in case no one comes up with a really convincing argument against it.

**Comment by nilsbeyer** [comment]
I nominate "Peter J" as the owner of the "Top Hat of Good Karma".

**Comment by Snark1994** [comment]
It was done as early as possible, which today happened to be later than 15 hours after the end of the voting day; I was busy.

I announced on WhatsApp that the vote had passed, and asked Nils who he wanted to nominate.

**Comment by malte-l** [comment]
I am not referring to the 15 hours part, just that it was not announced at all. WhatsApp is irrelevant, as that is not an announcement to all players. In particular not every player can check what came first etc. The merge should be reverted and the procedure followed properly. If you disagree, I invoke judgment.

**Comment by malte-l** [comment]
Also announcing whether the vote had passed or not is also insufficient, by rule 120 it has to be announced how every individual player voted.

**Comment by Snark1994** [comment]
Can Nils just comment on the github now, stating to whom he wishes to give the hat?

**Comment by malte-l** [comment]
You should first announce the results of the vote as required by rule 120...

**Comment by Snark1994** [comment]
Oh, apologies: I was under the impression I had done so (I had certainly typed out a comment doing so). I can't find this now, so presumably I didn't make the comment.

Everyone voted to approve apart from Malte. This gives me (302 - 291) * 5 / 6 = 9 points and Malte 10 points. Peter is now the owner of the THoGK and gains 4 points.

**Comment by malte-l** [comment]
ok, I still think technically N. would need to do the nomination with the hat again now to make it valid, but I won't contest implementing the change of ownership nevertheless. (so I retract the invocation of judgment)

**Comment by Snark1994** [comment]
Just to confirm, you're happy with us proceeding with the `master` branch as it currently is?

**Pull request merged**